pkg/detectors/aws/aws.go

```diff
@@ -47,6 +51,11 @@ func New(opts ...func(*scanner)) *scanner {
 		opt(scanner)
 	}
 
+	scanner.credsCache = memory.New(
+		memory.WithExpirationInterval(1*time.Hour),
```

Comment:
These are just placeholders for now, we could consider passing these in as args to `New`.

Reply:
Or use some default value.

Reply:
How big could the cache conceivably get in a given time frame?

Reply:
That's a good question. It would really depend on what was being scanned. We can run some tests to scan some repos and record some metrics to see what the size of this cache grows to be. Alternatively, we could use a LRU cache where we fix the size if needed.